src/makeCalculator.js

```diff
@@ -1,10 +1,53 @@
 'use strict';
 
-/**
+/*
  * @return {object}
  */
+
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    operate(callback, n = this.result) {
+      return callback.call(this, n);
+    },
+
+    add(n) {
+      this.result += n;
+
+      return this;
+    },
+
+    subtract(n) {
+      this.result -= n;
+
+      return this;
+    },
+
+    multiply(n) {
+      this.result *= n;
+
+      return this;
+    },
+
+    divide(n) {
+      if (n === 0) {
+        throw new Error('You can not divide on 0');
+      }
+
+      this.result /= n;
+
+      return this;
+    },
+
+    reset(n) {
```

Comment:
The `reset` method has an unused parameter `n`. Since this parameter is not utilized within the method, it should be removed to avoid confusion and improve code clarity.